go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -29,23 +31,23 @@ import (
 
 func (bnc *BaseNetworkController) allocatePodIPs(pod *kapi.Pod,
 	annotations *util.PodAnnotation, nadName string) (expectedLogicalPortName string, err error) {
-	switchName, err := bnc.getExpectedSwitchName(pod)
+	switchNames, err := bnc.getSwitchNames(pod)
```

Comment:
Please note that this method should execute as well for remote pods with an original switch name that is local to the zone.